lib/puma/cluster/worker.rb

```diff
@@ -14,7 +14,7 @@ class Cluster < Puma::Runner
     class Worker < Puma::Runner # :nodoc:
       attr_reader :index, :master
 
-      def initialize(index:, master:, launcher:, pipes:, server: nil)
+      def initialize(index:, master:, launcher:, pipes:, app: nil)
```

Comment:
There are two options here:
1. Continue to accept `server` and simply ensure that `@app` is initialized with `@server.app` if so. Then, calling `start_server` in `#spawn_worker` will use the preloaded worker 0 app to start a new server which we can give to the new worker. This would then mirror the `Cluster#spawn_worker` implementation.
2. Let the worker consistently be responsible for starting its own server, and instead accept the preloaded `app` where needed, which is what we really intend to preload. This makes the whole inheritance chain much easier to grok imo, introduces a clear separation of responsibilities.